src/components/Carousel.tsx

```diff
@@ -1,18 +1,180 @@
 import React from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[],
+};
+
+type State = {
+  step: number,
+  frameSize: number,
+  itemwidth: number,
+  animationDuration: number,
+  move: number,
+};
+
+class Carousel extends React.Component<Props, State> {
+  state: State = {
+    step: 3,
+    frameSize: 3,
+    itemwidth: 130,
+    animationDuration: 1000,
+    move: 0,
+  };
+
+  previousSlide = () => {
+    this.setState((currentState) => {
+      const scroll = currentState.move - (currentState.step * currentState.itemwidth);
+
+      return {
+        move: scroll < 0 ? 0 : scroll,
+      };
+    });
+  };
+
+  getMinScroll = () => {
+    const { images } = this.props;
+    const { step, itemwidth } = this.state;
+
+    return (images.length * itemwidth) + (step * itemwidth);
+  };
+
+  getMaxScroll = () => {
+    const { images } = this.props;
+    const { step, itemwidth } = this.state;
+
+    return (images.length * itemwidth) - (step * itemwidth);
+  };
+
+  nextSlide = () => {
+    this.setState((currentState) => {
+      const scroll = currentState.move + (currentState.step * currentState.itemwidth);
+
+      return {
+        move: scroll >= this.getMaxScroll() ? this.getMaxScroll() : scroll,
+      };
+    });
+  };
+
+  render() {
+    const { images } = this.props;
+    const {
+      step,
+      frameSize,
+      itemwidth,
+      animationDuration,
+      move: marginLeft,
+    } = this.state;
+
+    return (
+      <div className="content">
+        <div className="carousel">
+          <div className="carousel__wrapper">
+            <button
+              type="button"
+              className="btn btn--prev"
+              disabled={this.state.move <= 0}
+              onClick={this.previousSlide}
+            >
+              &#10148;
+            </button>
+            <div className="carousel__container" style={{ width: `${frameSize * itemwidth}px` }}>
+              <ul className="carousel__list" style={{ transform: `translateX(${-marginLeft}px)`, transition: `${animationDuration}ms`, width: `${images.length * itemwidth}px` }}>
```

Comment:
```suggestion
              <ul
                className="carousel__list"
                style={{ transform: `translateX(${-marginLeft}px)`, transition: `${animationDuration}ms`, width: `${images.length * itemwidth}px` }}
              >
```
Тут все ж краще перенести пропси кожен на новий рядок

Reply:
дякую. випрвив.